lib/DDGC/Web/Controller/InstantAnswer.pm

```diff
@@ -937,32 +681,29 @@ sub remove_edit {
     $ia->update({updates => $edits});
 }
 
-# given a result set, remove 
-# all the entries from the updates column
-sub remove_edits {
-    my($ia) = @_;
- 
-    $ia->update({updates => ''});
+# update the instant answer table
+sub update_ia {
+    my ($d,$ia, $field, $value) = @_;
+    $ia->update({$field => $value});
 }
 
-# given the IA name return the data in the updates
-# column as an array of hashes
-sub get_edits {
-    my ($d, $id) = @_; 
-
-    my $ia_result = $d->rs('InstantAnswer')->find($id);
-    my $edits;
-
-    try{
-        my $column_updates = $ia_result->get_column('updates');
-        $edits = $column_updates? from_json($column_updates) : undef;
-    }catch{
-        return;
-    };
-
-    return $edits;
+# given a result set and a field name, remove all the
+# entries for that field from the updates column
+sub remove_edits {
+    my($d, $ia, $field) = @_;   
+    # delete all entries from updates with field
+    my $edits = $d->rs('InstantAnswer::Updates')->search({ instant_answer_id => $ia->id, field => $field });
+    $edits->delete();
 }
 
+# given the IA name return a result set of all edits
+sub get_all_edits {
+    my ($d, $id) = @_;
+    warn "Getting edits for $id" if debug;
+    my @edits = $d->rs('InstantAnswer::Updates')->search( {instant_answer_id => $id} );
```

Comment:
...This line is probably causing the DB error, because it says "column me.instant_answer_id does not exist"